api/go.mod

```diff
@@ -0,0 +1,30 @@
+module github.com/openshift/hypershift/api
+
+go 1.21
+
+require (
+	github.com/openshift/api v0.0.0-20240108220607-355cd2586ec4
```

Comment:
Are you sure these are all the newest versions? I know the k8s ones have a newer version. Just trying to prevent more dependabot PRs :)

Reply:
humm not fully sure. The only dependency I updated in purpose is in the `hack/tools/go.mod` is code-generator